octoprint_mrbeam/filemanager/file_storage.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+import os
+import time
+
+
+from octoprint.filemanager.storage import LocalFileStorage, StorageError
+from octoprint_mrbeam.mrb_logger import mrb_logger
+
+
+class MrBeamFileStorage(LocalFileStorage):
+    def __init__(self, basefolder, create=False):
+        self._logger_mrb = mrb_logger(
+            "octoprint.plugins.mrbeam.filemanager.mrbFileStorage"
+        )
+
+        LocalFileStorage.__init__(self, basefolder, create)
+
+    def remove_multiple_files(self, files):
+        metadata_to_remove = {}
+        f_count = 0
+        ts_start = time.time()
+
+        for path in files:
+            path, name = self.sanitize(path)
+
+            file_path = os.path.join(path, name)
+            if not os.path.exists(file_path):
+                continue
+            if not os.path.isfile(file_path):
+                self._logger_mrb.warn(
+                    "remove_multiple_files(): {name} in {path} is not a file -> skipping".format(
+                        name=name, path=path
+                    )
+                )
+
+            try:
+                os.remove(file_path)
+                f_count += 1
+            except Exception as e:
+                self._logger_mrb.error(
+                    "Could not delete {name} in {path}, cause: {cause}".format(
+                        name=name, path=path, cause=e
+                    )
+                )
+                continue
+
+            # collect for metadata removal
+            if path not in metadata_to_remove:
+                metadata_to_remove[path] = []
+            metadata_to_remove[path].append(name)
+
+        self._remove_metadata_multiple_entry(metadata_to_remove)
+
+        self._logger_mrb.info(
+            "remove_multiple_files(): Deleted %s/%s files in %s paths in %.2f sec",
+            f_count,
+            len(files),
+            len(metadata_to_remove.keys()),
+            time.time() - ts_start,
+        )
+
+    def _remove_metadata_multiple_entry(self, entries):
+        for path in entries.keys():
+            with self._get_metadata_lock(path):
+                metadata = self._get_metadata(path)
+
+                for name in entries[path]:
+                    if not name in metadata:
+                        continue
+
+                    if "hash" in metadata[name]:
```

Comment:
```
from octoprint_mrbeam.util import dict_get
hash = dict_get(metadata, [name, "hash"])
if hash: # or is hash is not None:
    # do stuff
```

Reply:
OP copy-past code.

Reply:
Then please document the function:
- declare it's a copy-pasta from function x
- why it is a copy-pasta

Reply:
Makes it easier to understand later on and to create a PR for OctoPrint